apps/scotty/collect.go

```diff
@@ -280,12 +280,12 @@ type memoryCheckerType interface {
 	Check()
 }
 
-func createCloudHealthWriter(path string) *cloudhealth.Writer {
+func newCloudHealthWriter(reader io.Reader) (interface{}, error) {
```

Comment:
I will rework the dynconfig API so that it can accept a function that returns any specific type and an error.